torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -287,6 +295,22 @@ bool FunctionSignature::parse(PyObject* args, PyObject* kwargs, PyObject* dst[],
         arg_pos++;
         continue;
       } else {
+        if (allow_varargs_intlist && arg_pos == 0) {
+          for (int j = arg_pos; j < nargs; ++j) {
```

Comment:
The check only applies to the varargs case: e.g. `view(1, 2, 3)`. It doesn't check the tuple case of `view((1, 2, 3))`. In that case we still need to check the contents in the unpack code.

Reply:
We do already check the contents (although the error message isn't as nice as above).
```
>>> a.view((3,2,1.5))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Could not unpack long
```

Reply:
It'd be good to have high quality error messages for both cases

I don't see the need to do the check of contents here. If `IntList` binds to any tuple (with the contents check deferred) then `IntList` + varargs can bind to any sequence of arguments (with the contents check deferred).

Reply:
That should work.  We might want to revisit deferring the checking in general because it makes it more difficult to reason about (i.e. it wasn't immediately obvious in the previous PR that we weren't checking tensorlists properly).